assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,307 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:EditBuildConfigController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('EditBuildConfigController', function ($scope, $routeParams, DataService, ProjectsService, $filter, ApplicationGenerator, Navigate, $location, AlertMessageService) {
+
+    $scope.projectName = $routeParams.project;
+    $scope.buildConfig = null;
+    $scope.alerts = {};
+    $scope.emptyMessage = "Loading...";
+    $scope.options = {};
+
+    $scope.buildFromTypes = [
+      {
+        "id": "ImageStreamTag",
+        "title": "Image Stream Tag"
+      },
+      {
+        "id": "DockerImage",
+        "title": "Docker Image Link"
+      },
+    ];
+
+    $scope.breadcrumbs = [
+      {
+        title: "Builds",
+        link: "project/" + $routeParams.project + "/browse/builds"
+      },
+      {
+        title: $routeParams.buildconfig,
+        link: "project/" + $routeParams.project + "/browse/builds/" + $routeParams.buildconfig
+      },
+      {
+        title: "Edit"
+      }
+    ];
+
+    $scope.buildFrom = {
+      projects: [],
+      imageStreams: [],
+      tags: {},
+    }
+    $scope.pushTo = {
+      projects: [],
+      imageStreams: [],
+      tags: {},
+    }
+
+    $scope.triggers = {
+      "webhook": false,
+      "imageChange": false,
+      "configChange": false
+    };
+
+    var watches = [];
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        DataService.get("buildconfigs", $routeParams.buildconfig, context).then(
+          // success
+          function(buildConfig) {
+            $scope.buildConfig = buildConfig;
+            $scope.updatedBuildConfig = angular.copy($scope.buildConfig);
+            $scope.buildStrategy = $filter('buildStrategy')($scope.updatedBuildConfig);
+            $scope.envVars = $filter('envVarsPair')($scope.buildStrategy.env);
+
+            angular.forEach($scope.buildConfig.spec.triggers, function(value, key) {
+              switch (value.type) {
+                case "Generic":
+                  break;
+                case "GitHub":
+                  $scope.triggers["webhook"] = true;
+                  break;
+                case "ImageChange":
+                  $scope.triggers["imageChange"] = true;
+                  break;
+                case "ConfigChange":
+                  $scope.triggers["configChange"] = true;
+                  break;
+              }
+            });
+            $scope.pickedTriggers = $scope.triggers;
+            
+
+            var buildFrom = $scope.buildStrategy.from;
+            var testFrom = $filter('imageObjectRef')($filter('buildStrategy')($scope.buildConfig).from, buildConfig.metadata.namespace)
```

Comment:
Multiple calls to `$filter` on one line are hard to read. I'd probably just assign the filters to variables above.